include/aws/cryptosdk/materials.h

```diff
@@ -472,12 +472,31 @@ struct aws_cryptosdk_keyring_vt {
         enum aws_cryptosdk_alg_id alg);
 };
 
+/**
+ * Constant time check of data-structure invariants for struct aws_cryptosdk_keyring_vt.
+ */
+AWS_CRYPTOSDK_STATIC_INLINE bool aws_cryptosdk_keyring_vt_is_valid(const struct aws_cryptosdk_keyring_vt *vtable) {
+    return AWS_OBJECT_PTR_IS_READABLE(vtable) && (vtable->vt_size == sizeof(struct aws_cryptosdk_keyring_vt)) &&
```

Comment:
are we guaranteed its that sizeof? Can it be overloaded to a larger size?

Reply:
The documentation says:
```c
struct aws_cryptosdk_keyring_vt {
    /**
     * Always set to sizeof(struct aws_cryptosdk_keyring_vt).
     */
    size_t vt_size;
...
```

Reply:
Add a comment to the is_valid saying that, and 🚢 it